sql/src/main/java/io/crate/metadata/sys/ClassifiedMetrics.java

```diff
@@ -27,24 +27,41 @@
 import org.HdrHistogram.Histogram;
 
 import java.util.Iterator;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
 
-public class ClassifiedHistograms implements Iterable<ClassifiedHistograms.ClassifiedHistogram> {
+public class ClassifiedMetrics implements Iterable<ClassifiedMetrics.Metrics> {
 
     private static final long HIGHEST_TRACKABLE_VALUE = TimeUnit.MINUTES.toMillis(10);
     private static final int NUMBER_OF_SIGNIFICANT_VALUE_DIGITS = 3;
 
-    private final ConcurrentHashMap<Classification, ConcurrentHistogram> histograms = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<Classification, Metrics> metrics = new ConcurrentHashMap<>();
 
-    public static class ClassifiedHistogram {
+    public static class Metrics {
 
-        private Histogram histogram;
-        private Classification classification;
+        private final Classification classification;
+        private final ConcurrentHistogram histogram;
+        private final LongAdder sumOfDurations = new LongAdder();
 
-        ClassifiedHistogram(Histogram histogram, Classification classification) {
-            this.histogram = histogram;
+        /**
+         * @param classification
+         * @param highestTrackableValue          represents the highest value to track
+         * @param numberOfSignificantValueDigits the precision to use
+         */
+        Metrics(Classification classification, final long highestTrackableValue, final int numberOfSignificantValueDigits) {
             this.classification = classification;
+            this.histogram = new ConcurrentHistogram(highestTrackableValue, numberOfSignificantValueDigits);
+        }
+
+        void recordValue(long duration) {
+            // We use start and end time to calculate the duration (since we track them anyway)
+            // If the system time is adjusted this can lead to negative durations
+            // so we protect here against it.
+            long safeDuration = Math.min(Math.max(0, duration), HIGHEST_TRACKABLE_VALUE);
```

Comment:
Sorry I didn't catch this earlier, but now I'm wondering if we should apply the cap (`HIGHEST_TRACKABLE_VALUE`) only for the histogram but not for the `sumOfDurations`.  Thoughts?

Reply:
why are we applying this cap for the histogram?